### ISSUE TITLE
Add partition, validDocId as the derived column

### ISSUE THREAD

**Issue by snleee** [apache/pinot#11580]
We currently support some derived columns(e.g. $segmentName , $docId ).

It will be useful to add $partitionValues, $isValidDocId

**Comment by xiangfu0** [comment]
+1

**Comment by ArnavBalyan** [comment]
Hi I would like to work on this issue, can you please assign it to me. Thanks

**Comment by Jackie-Jiang** [comment]
@snleee let's clarify the content of these derived columns:
- For `$partitionValues`, are you referring to the partition id of the segment?
- For `$isValidDocId`, is that for debugging purpose when skipping the upsert?

**Comment by ArnavBalyan** [comment]
Hi @snleee gentle reminder if you could please give clarity on this. Thanks!

**Comment by deemoliu** [comment]
@ArnavBalyan can you write a design doc before implementation on this? 
I think you can split the implementation for `$partitionValues` and `$isValidDocId`.

- `$partitionValues` is straight forward because it won't be updated by later events. 
- For `$isValidDocId`,  i am curious about the design since validDocIds will be updated via partitionUpsertDataManager.

**Comment by snleee** [comment]
1. `$partitionValues` are the ones partition id of the segment. This information is already getting populated to segment metadata as `partitionValues` when the customer partition function is configured.

2. `$isValidDocId` is for debugging purpose when the upsert feature is enabled.

**Comment by Jackie-Jiang** [comment]
@snleee 

> 1. `$partitionValues` are the ones partition id of the segment. This information is already getting populated to segment metadata as `partitionValues` when the customer partition function is configured.

Do we want to return the segment level partition values, or the per record partition id?

> 2. `$isValidDocId` is for debugging purpose when the upsert feature is enabled.

This is not really useful. When upsert is enabled, all records will have this field `true`.

**Comment by snleee** [comment]
@Jackie-Jiang Good point. Per record partition id would be better as this is getting appended per each row.

**Comment by MeihanLi** [comment]
@ArnavBalyan do we have any progress on the issue? Are you still working on it?

**Comment by ArnavBalyan** [comment]
Hi yes, I have some progress will update here shortly thanks
